src/matteruser.js

```diff
@@ -106,8 +108,19 @@ class Matteruser extends Adapter {
       process.exit(1);
     }
 
-    this.client = new MatterMostClient(mmHost, mmGroup, {wssPort: mmWSSPort, httpPort: mmHTTPPort, pingInterval: 30000, httpProxy: mmHTTPProxy});
+    this.client = new MatterMostClient(mmHost, mmGroup, {
+      wssPort: mmWSSPort, httpPort: mmHTTPPort, pingInterval: 30000, httpProxy: mmHTTPProxy
+    });
+
+    this.declareCallbacks();
+    if (mmAccessToken != null) {
+      return this.client.tokenLogin(mmAccessToken);
+    }
+
+    return this.client.login(mmUser, mmPassword, mmMFAToken);
+  }
 
+  declareCallbacks() {
```

Comment:
Extract callback declaration in dedicated function to avoid to break lint rule on maximum function lines